synapse/types.py

```diff
@@ -151,6 +155,10 @@ class DomainSpecificString(namedtuple("DomainSpecificString", ("localpart", "dom
         'domain' : The domain part of the name
     """
 
+    __metaclass__ = abc.ABCMeta
```

Comment:
I think this is the Python 2 way of declaring metaclasses?

Reply:
That's what we do elsewhere. I'm not sure if its "better" to move it up?

Reply:
I don't have a strong opinion. 👍